jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -608,6 +608,10 @@ public boolean onIdleExpired(TimeoutException timeout)
         if (_httpChannel.getRequest() == null)
             return true;
         Runnable task = _httpChannel.onIdleTimeout(timeout);
+
+        // TODO should we run the task directly here, even though that may block the scheduler?
```

Comment:
That sounds like something that `Invocable` is meant to solve, i.e. something like:

```java
if (Invocable.getInvocationType(task) == BLOCKING)
  getExecutor().execute(task);
else
  task.run();
```

Reply:
I think we need to look at encoding that logic into a `ExceptionUtil.mustExecute(Executor, Runnable)` method that will ensure execptional tasks are run one way or another.